submarine-server/server-core/src/main/java/org/apache/submarine/server/s3/Client.java

```diff
@@ -45,30 +47,57 @@
 /**
  * S3(Minio) default client
  */
-public class Client {
+public enum Client {
+  DEFAULT(SubmarineConfiguration.getInstance().getString(SubmarineConfVars.ConfVars.SUBMARINE_S3_ENDPOINT)),
```

Comment:
I don't think it's necessary to add an endpoint attribute to the enum, which can't declare a variable number of endpoints and has to use the internal map to handle it.
```java
public enum Client {
  INSTANCE;
  public static Client getInstance(String endpoint) {
    // ....
  }
}

Client.INSTANCE.getInstance("http://localhost:9000");
```

Reply:
Hi @cdmikechen ,
Not sure what you mean here as well. Are you suggesting not initiating the enum variable?

Reply:
When using an enum as a factory to get a single instance object, using a limited number of enum definitions does not achieve the final effect of a dynamic factory